agent/agent.go

```diff
@@ -298,6 +298,44 @@ func (a *Agent) withSession(ctx context.Context, fn func(session *session) error
 	}
 }
 
+// AttachNetwork sends a container attach network request towards a
```

Comment:
@mrjana The main issue is that we are coupling network attachment to the agent type: https://github.com/mrjana/docker/commit/db39e4d435e28a3cca67f124ab906cd413f6d207#diff-9cb7240401da846ac7c2252759538e9aR1213.

Let's have the agent return a resource allocator interface that can be passed to the particular component. I want to be able to change the agent exported functions without having to change daemon code.